color_palette/index.html

```diff
@@ -0,0 +1,72 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <link rel="stylesheet" href="styles.css" />
+    <title>Document</title>
+  </head>
+  <body>
+    <div class="container-main">
+      <div class="container-big-1">
+        <div class="container-1">
+          <div class="colorbox main box-1" data-number="1">
+            <input type="color" name="1" id="color1" value="#000000" />
+          </div>
+
+          <div class="colorbox main box-2" data-number="2">
+            <input type="color" name="2" id="color2" value="#000000" />
+          </div>
+
+          <div class="colorbox main box-3" data-number="3">
+            <input type="color" name="3" id="color3" value="#000000" />
+          </div>
+        </div>
+
+        <div class="container-2">
+          <div class="colorbox neut box-4" data-number="4">
+            <input type="color" name="4" id="color4" value="#000000" />
+          </div>
+
+          <div class="colorbox neut box-5" data-number="5">
+            <input type="color" name="5" id="color5" value="#000000" />
+          </div>
+        </div>
+      </div>
+
+      <div class="container-big-2">
+        <div class="container-3">
+          <div class="colorbox acc box-6" data-number="6">
+            <input type="color" name="6" id="color6" value="#000000" />
+          </div>
+
+          <div class="colorbox acc box-7" data-number="7">
+            <input type="color" name="7" id="color7" value="#000000" />
+          </div>
+
+          <div class="colorbox acc box-8" data-number="8">
+            <input type="color" name="8" id="color8" value="#000000" />
+          </div>
+
+          <div class="colorbox acc box-9" data-number="9">
+            <input type="color" name="9" id="color9" value="#000000" />
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <script>
+      const inputs = document.querySelectorAll("input");
+      function changeColor() {
+        colorbox = document.querySelector(
+          `.colorbox[data-number="${this.name}"]`
+        );
```

Comment:
Interesting that you use `this` here. Where did you learn it? It works in this case, but usually, when handling events, it's used the argument `event` that is passed to `changeColor` instead.  `function changeColor(event) { .... }` In this case. `event.target` is the same as `this`, so `event.target.name` is the same as `event.name`.

Also, always remember to declare a variable using var, let or const. `const colorbox = ...`.

Reply:
I have used another example from JavaScript 30 challenge. It was a pressing key code, but I just reformatted it to work this way. I will try to make it using changeColor as well.